src/vmm/src/cpu_config/x86_64/custom_cpu_template.rs

```diff
@@ -144,12 +144,9 @@ pub struct CustomCpuTemplate {
 }
 
 impl CustomCpuTemplate {
-    /// Get a list of MSR indices that are modified by the CPU template.
-    pub fn get_msr_index_list(&self) -> Vec<u32> {
-        self.msr_modifiers
-            .iter()
-            .map(|modifier| modifier.addr)
-            .collect()
+    /// Get an iterator of MSR indices that are modified by the CPU template.
+    pub fn msr_index_iter(&self) -> impl ExactSizeIterator<Item = u32> + '_ {
```

Comment:
I don't have strong preference, but if you want to reduce unneeded allocation, how about making it return `impl ExactSizeIterator<Item = &u32>`? By this we can remove some `.copied()` (e.g., `vcpu.msrs_to_save.iter().copied()`, `self.msrs_to_save.iter().copied()`, `msrs_to_dump.as_slice().iter().copied()`).

Reply:
I don't like `.copied()` business as well, but this does not do any allocations. Here we are talking about `&u32` and `u32` which is actually cheaper to copy than to have references (`u32` is 32 bits, `&u32` is `usize` so 64 bits).